assets/src/components/vehicleIcon.tsx

```diff
@@ -173,251 +206,255 @@ export const viewBox = ({
   return { left, top, width, height }
 }
 
-export const VehicleIconSvgNode = ({
-  size,
-  orientation,
-  label,
-  variant,
-  status,
-  alertIconStyle,
-}: Props): ReactElement<SVGElement> => {
-  const [{ userSettings }] = useContext(StateDispatchContext)
-
-  status = status || "plain"
-  variant = variant && variant !== "_" ? variant : undefined
-  // ghosts can't be drawn sideways
-  if (
-    status === "ghost" &&
-    (orientation === Orientation.Left || orientation === Orientation.Right)
-  ) {
-    orientation = Orientation.Up
-  }
-  const classes: string[] = [
-    "m-vehicle-icon",
-    `m-vehicle-icon${sizeClassSuffix(size)}`,
-    alertIconStyle === AlertIconStyle.Highlighted
-      ? "m-vehicle-icon--highlighted"
-      : "",
-  ].concat(statusClasses(status, userSettings.vehicleAdherenceColors))
-  return (
-    <g className={className(classes)}>
-      {label ? (
-        <Label size={size} orientation={orientation} label={label} />
-      ) : null}
-      {status === "ghost" ? (
-        <Ghost size={size} variant={variant} />
-      ) : (
-        <Triangle size={size} orientation={orientation} />
-      )}
-      {variant ? (
-        <Variant
-          size={size}
-          orientation={orientation}
-          variant={variant}
-          status={status}
-        />
-      ) : null}
-
-      {alertIconStyle ? (
-        <AlertCircleIcon
-          size={size}
-          orientation={orientation}
-          status={status}
-          alertIconStyle={alertIconStyle}
-        />
-      ) : null}
-    </g>
-  )
-}
+export const VehicleIconSvgNode = React.memo(
+  ({
+    size,
+    orientation,
+    label,
+    variant,
+    status,
+    alertIconStyle,
+  }: Props): ReactElement<SVGElement> => {
+    const [{ userSettings }] = useContext(StateDispatchContext)
```

Comment:
Did we determine exactly how this will play along with the memoization? I guess `userSettings` won't change very often, though the rest of `StateDispatchContext` might.

Reply:
Per the `React.memo` docs:

> If your function component wrapped in React.memo has a useState, useReducer or useContext Hook in its implementation, it will still rerender when state or context change.

Reply:
Will that happen whenever anything from `StateDispatchContext` changes, or just `userSetings`? I wonder if we can get even more of an improvement by moving the `useContext` higher up, extracting `userSettings` from the state, and passing it down as a prop.

Reply:
> Will that happen whenever anything from StateDispatchContext changes, or just userSetings?

That's an interesting question...I wonder if Stackoverflow would know, not sure how to phrase the search though. In any case we can probably lift the `useContext`, will give it a try.